### PR TITLE
Create custom react hook for using metamask wallet to login and connect

### PR DESCRIPTION
**Pull request by clemencegoh**
## What
Custom react hook for ease of connecting to metamask.

## Usage
```
	const [{ name, address }, setUserName, connectToMetamask] = useMetamaskWallet();
```

- Destructure the object to receive name, address, and chainID if needed.

### CODE REVIEW

File: src/components/ExchangeCard.react.tsx [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-    <div>Swap</div>
-    <div>From</div>
-    <input></input>
-    <br />
-    <button onClick={() => {}}>&lt;&gt;</button>
-    <div>To</div>
-    <input></input>
-    <button onClick={() => {}}>Connect Wallet</button>
-  </div>
+export default function ExchangeCard(): React.ReactElement {
```
Comment (clemencegoh) [inline src/components/ExchangeCard.react.tsx:3]: changes here caused by prettier

Comment (LeaKyJar) [reply src/components/ExchangeCard.react.tsx:3]: Oof, should add that extension as well

**Review by LeaKyJar — APPROVED** [review]
To clarify, we're storing wallet details in local storage since non-prod, yea?

**Comment by clemencegoh** [comment]
> To clarify, we're storing wallet details in local storage since non-prod, yea?

We're storing a map of wallet address => user preferences like display name, theme, etc.
Since Metamask lives on the local browser environment, it's guaranteed that every user will have unique wallet addresses (therefore eliminating our need to allow users to log in manually)

**Pull request merged**